recipes/qhull/all/test_package/CMakeLists.txt

```
@@ -1,20 +1,32 @@
 cmake_minimum_required(VERSION 3.15)
-project(test_package LANGUAGES C)
+project(test_package)
 
 find_package(Qhull REQUIRED CONFIG)
 
+# Select source and target depending on reentrant flag
 if(QHULL_REENTRANT)
-    add_executable(${PROJECT_NAME} test_package_r.c)
+    set(TEST_SRC test_package_r.c)
     if(TARGET Qhull::qhull_r)
-        target_link_libraries(${PROJECT_NAME} PRIVATE Qhull::qhull_r)
+        set(TEST_LIB Qhull::qhull_r)
     else()
-        target_link_libraries(${PROJECT_NAME} PRIVATE Qhull::qhullstatic_r)
+        set(TEST_LIB Qhull::qhullstatic_r)
     endif()
 else()
-    add_executable(${PROJECT_NAME} test_package.c)
+    set(TEST_SRC test_package.c)
     if(TARGET Qhull::libqhull)
-        target_link_libraries(${PROJECT_NAME} PRIVATE Qhull::libqhull)
+        set(TEST_LIB Qhull::libqhull)
     else()
-        target_link_libraries(${PROJECT_NAME} PRIVATE Qhull::qhullstatic)
+        set(TEST_LIB Qhull::qhullstatic)
     endif()
 endif()
+
+# Main C test
+add_executable(${PROJECT_NAME} ${TEST_SRC})
+target_link_libraries(${PROJECT_NAME} PRIVATE ${TEST_LIB})
+
+# Optional C++ test (only with static builds)
+if(TARGET Qhull::qhullcpp)
+    add_executable(${PROJECT_NAME}_cpp test_package_cpp.cpp)
+    target_link_libraries(${PROJECT_NAME}_cpp PRIVATE ${TEST_LIB} Qhull::qhullcpp)
+    set_property(TARGET ${PROJECT_NAME}_cpp PROPERTY CXX_STANDARD 11)
```

```suggestion
    target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_11)
```
I would not enforce C++11, but let settings.compiler.cppstd to decide